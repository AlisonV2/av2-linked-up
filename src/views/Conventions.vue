<template>
<div class="container">
  <div class="row convention-row" data-test="conventions">
    <app-title>Upcoming events</app-title>
    <div class="col-12 col-lg-4" v-for="event in events" :key="event.id">
      <EventItem :event="event" />
    </div>
  </div>
</div>
</template>

<script>
import EventItem from '@/components/admin/EventItem';
import * as Sentry from '@sentry/vue';

/**
 * @exports Conventions
 * @type {Page}
 */
export default {
  name: 'Events',
  components: {
    EventItem,
  },
  data() {
    return {
      events: [],
    };
  },
  created() {
    this.$store
      .dispatch('getEvents')
      .then(() => {
        this.events = this.$store.getters.getEvents;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>

<style lang="scss">
.convention-row {
  width: 100%;
}
</style>
