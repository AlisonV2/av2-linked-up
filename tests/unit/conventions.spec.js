import { shallowMount } from '@vue/test-utils';
import Conventions from '@/views/Conventions.vue';
import AppTitle from '@/components/app/Title.vue'

/**
 * @module ConventionsPageTest
 */
describe('Conventions.vue', () => {
  it('Check if header is relevant', () => {
    const component = shallowMount(Conventions, {
      global: {
        components: {
          'app-title': AppTitle,
        },
      },
    });
    const title = component.findComponent(AppTitle);
    expect(title.exists()).toBe(true);
  });
});
