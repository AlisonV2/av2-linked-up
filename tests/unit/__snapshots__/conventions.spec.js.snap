// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conventions.vue Conventions Snapshot 1`] = `
<div
  class="row convention-row"
  data-test="conventions"
>
  <app-title-stub />
</div>
`;
